README.md

```diff
@@ -440,6 +440,31 @@ curl "http://localhost:9091/api/v1/seriesMetadata?filter=http&type=counter"
 }
 ```
 
+### Push Query Data using API
+
+`POST /api/v1/query/push` Push query data to the server, for cases when you have other means of collecting PromQL queries stats.
+
+#### Request Format
+
+```jsonnet
+[
+  {
+    "type": "instant",                     // REQUIRED Type of the query, one of `range` or `instant`
+    "query_param": "up{job='foobar'}",     // REQUIRED The PromQL expression of the query
+    "time_param": "2024-01-15T10:30:00Z",  // The time when the query was evaluated, only for type `instant` in RFC 3339 format
+    "duration": 250000000,                 // Duration of the query in nanoseconds
+    "status_code": 200,                    // REQUIRED Valid HTTP status code, see https://go.dev/src/net/http/status.go
+    "body_size": 512,                      // Response body size in bytes
+    "step": 15,                            // REQUIRED(for type `range`) Range query step in seconds
+    "start": "2024-01-15T10:30:00Z",       // REQUIRED(for type `range`) Range query start time in RFC 3339 format
+    "end": "2024-01-15T10:30:00Z",         // REQUIRED(for type `range`) Range query end in RFC 3339 format
```

Comment:
I actually don't like supporting just the RFC 3339 format, but otherwise we would have to use the separate struct with custom unmarshaling.

Reply:
I see, I'm mostly relying on Prometheus standards, which makes sense on native proxy, but eventually for push it might not make.

Maybe we can wait and get more feedback, wdyt @FUSAKLA ?

Reply:
Sorry for the delay… sure, if you are OK with that, let's leave it like that for the time being.

We can add support for the timestamp format which would require the separate struct we talked about, but that wouldn't break the API, just add another supported format so should be good.